src/super_gradients/training/datasets/detection_datasets/detection_dataset.py

```diff
@@ -301,20 +305,24 @@ def get_sample(self, index: int) -> Dict[str, Union[np.ndarray, Any]]:
         :param index:   Image index
         :return:        Sample, i.e. a dictionary including at least "image" and "target"
         """
-        img = self.get_resized_image(index)
-        annotation = self.annotations[index]
+        img = self.get_rescaled_image(index)
+        annotation = deepcopy(self.annotations[index])
         return {"image": img, **annotation}
 
-    def get_resized_image(self, index: int) -> np.ndarray:
+    def get_rescaled_image(self, index: int) -> np.ndarray:
         """
-        Get the resized image at a specific sample_id, either from cache or by loading from disk, based on self.cached_imgs
+        Get the rescaled image (i.e. either width or height reaches its input_dim) at a specific sample_id,
+        either from cache or by loading from disk, based on self.cached_imgs_padded
         :param index:  Image index
-        :return:       Resized image
+        :return:       rescaled image
         """
         if self.cache:
-            return self.cached_imgs[index].copy()
+            padded_image = self.cached_imgs_padded[index]
+            rescaled_height, rescaled_width = self.annotations[index]["rescaled_img_shape"]
+            rescaled_image = padded_image[:rescaled_height, :rescaled_width, :]
```

Comment:
Is there ever going to be padding here?

Reply:
The cache does pad because it expects a fixed size tensor, it is done here https://github.com/Deci-AI/super-gradients/blob/6810c3c5888ae39b233a351ee459961d32bce3a3/src/super_gradients/training/datasets/detection_datasets/detection_dataset.py#L234-L235
So in the lines you commented we undo this padding, because otherwise it would affect the transforms